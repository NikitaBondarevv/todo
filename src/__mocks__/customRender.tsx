import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'

import { store } from 'store'

export const customRender = (Cpmponent: React.ReactNode) => render(
  <BrowserRouter>
    <Provider store={store}>
      {Cpmponent}
    </Provider>
  </BrowserRouter>
)