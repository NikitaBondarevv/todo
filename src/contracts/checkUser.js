import { request } from './request'

export const checkUser = async () => request.get('public/checkUser')
