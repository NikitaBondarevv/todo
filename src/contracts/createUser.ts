import { request } from './request'
import { IData } from 'interfaces/IData'

export const createUser = async (data: IData) => request.post('public/user', data)
