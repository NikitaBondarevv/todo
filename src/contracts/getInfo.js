import { request } from './request'

export const getInfo = async () => request.get('info')
