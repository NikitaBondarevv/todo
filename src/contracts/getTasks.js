export const getTasks = async () => {
  const response = await fetch('http://localhost:8086/tasks', { credentials: 'include' })
  
  return await response.json()
}
