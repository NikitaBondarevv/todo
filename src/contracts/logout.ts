import { request } from './request'

export const logout = async () => request.get('logout')
