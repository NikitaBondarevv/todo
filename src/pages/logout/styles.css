.logout {
  border: 1px solid #fff;
  box-shadow: 0 0 5px 1px rgb(0 0 0 / 50%);
  padding: 39px 29px 104px;
  background: rgb(7 21 147 / 50%);
  width: 777px;
  height: 383px;
  color: #fff;
  font: 24px/40px "MyriadProRegular";
  margin: 0 auto 100px;
  letter-spacing: 1px;
}

.error {
  color: red;
  font-size: 25px;
  margin: 0 0 0 5px;
}
