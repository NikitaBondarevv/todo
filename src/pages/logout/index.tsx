import { useEffect, useState } from 'react'

import { useAppDispatch } from 'store'
import { logout } from 'contracts/logout'
import { logout as logoutAction} from 'store/user'
import styles from './styles.css'


export const Logout = () => {
  const dispatch = useAppDispatch()
  const [error, setError] = useState('')

  const logoutHandler = async () => {
    try {
      await logout()
      dispatch(logoutAction())
    } catch (error) {
      setError('something went wrong, please try leater!')
    }
  }
  
  useEffect(() => {
    logoutHandler()
  }, [])

  return (
    <div className={styles.logout}>
      <span>
        User is signing out...
        <span className={styles.error}>
          {error}
        </span>
      </span>
    </div>
  )
}
