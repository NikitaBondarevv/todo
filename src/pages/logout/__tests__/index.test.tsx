import { screen } from '@testing-library/react'

import { logout } from 'contracts/logout'
import { customRender } from '__mocks__/customRender'
import { Logout } from '..'

jest.mock('contracts/logout', () => ({
  logout: jest.fn()
}))

describe('<Logout />', () => {
  const logoutMock = logout as jest.Mock

  test('should match snapshot', () => {
    const { asFragment } = customRender(<Logout />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call logout if don`t error', () => {
    customRender(<Logout />)

    expect(logout).toHaveBeenCalled()
  })

  test('should show error text if "logout" don`t work', async () => {
    const mockError = 'something went wrong, please try later!'

    logoutMock.mockRejectedValueOnce(mockError)
    customRender(<Logout />)

    await logout()

    expect(await screen.findByText(mockError)).toBeInTheDocument()
  })
})
