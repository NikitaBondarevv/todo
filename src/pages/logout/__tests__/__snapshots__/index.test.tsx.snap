// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="logout"
  >
    <span>
      User is signing out...
      <span
        class="error"
      />
    </span>
  </div>
</DocumentFragment>
`;
